### PR TITLE
[Core] Move CoordinateTransformationUtilities Methods to Public

### PR DESCRIPTION
**Pull request by sunethwarna**
**Description**
This PR moves `CoordinateTransformationUtils::LocalRotationOperatorPure` to public so it can be accessed without always checking the dimensionality specially in `elements` and `conditions` where dimensionality is defined by the template arguments.

**Changelog**
- Move `LocalRotationOperatorPure` to public interface.

### CODE REVIEW

File: kratos/utilities/coordinate_transformation_utilities.h [hunk 1]
```diff
@@ -23,6 +23,7 @@
 // kratos includes
 #include "includes/define.h"
 #include "includes/node.h"
+#include "includes/model_part.h"
```
Comment (philbucher) [inline kratos/utilities/coordinate_transformation_utilities.h:26]: ouch this is a heavy include
AFAIK this file is included in many elements no?

=> Why do you include it here?

Comment (sunethwarna) [reply kratos/utilities/coordinate_transformation_utilities.h:26]: I have a utilitiy which only uses geomerty as in this utility, then `ModelPart` becomes an incomplete type and compilation fails.

Comment (philbucher) [reply kratos/utilities/coordinate_transformation_utilities.h:26]: ok I checked, this is not included in elements

also the ModelPart is being used in there for a while already, then all good

**Review by philbucher — APPROVED** [review]

**Comment by sunethwarna** [comment]
@philbucher  thanks :)

**Pull request merged**